### ISSUE TITLE
Broken for BSD?

### ISSUE THREAD

**Issue by toddr** [akiym/Crypt-OpenSSL-Guess#3]

**Comment by toddr** [comment]
I'm about to issue a 0.29_03 build but it looks like openbsd is broken. Not sure of root cause but thought I'd open here to see if you have any ideas?

http://matrix.cpantesters.org/?dist=Crypt-OpenSSL-RSA%200.29_02

**Comment by akiym** [comment]
Perhaps It's Crypt::OpenSSL::RSA problem. OpenBSD is using LibreSSL instead of OpenSSL, so it doesn't provide RSA_get0_key and we should fix the patch of adapt to OpenSSL 1.1.0.

```c
#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined LIBRESSL_VERSION_NUMBER
    d = p_rsa->rsa->d;
#else
    RSA_get0_key(p_rsa->rsa, NULL, NULL, &d);
#endif
```

https://github.com/toddr/Crypt-OpenSSL-RSA/commit/5e7842169fe47f4cd63b2b4acca39d5f34febf8f

**Comment by toddr** [comment]
Yep that looks right sorry for the false alarm
